scientific-computing-with-python/learn-special-methods-by-building-a-vector-space/vectors.py

```diff
@@ -0,0 +1,96 @@
+class Vector:
+    space = 'R3'
+
+    def __init__(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+        self.norm = (x**2 + y**2 + z**2)**0.5
+
+    def __str__(self):
+        return str((self.x, self.y, self.z))
+    
+    def __repr__(self):
+        return f'Vector({self.x}, {self.y}, {self.z})'
+
+    def __add__(self, other):
+        if not isinstance(other, Vector):
+            return NotImplemented
+        x = self.x + other.x
+        y = self.y + other.y
+        z = self.z + other.z        
+        return Vector(x, y, z)
```

Comment:
I remembered another detail. It might be better to return the new instance of the _current_ class, instead of directly `Vector`. The difference is that if somebody uses `Vector` as a parent class, without changing method returning `Vector`, the child class will keep returning `Vector`, instead of the child class.

Therefore to make it more flexible, in case somebody uses `Vector` as a parent class:
```suggestion
        return self.__class__(x, y, z)
```